frontend/src/general/SnackbarMessage.jsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent, IconButton, Box } from "@material-ui/core";
+import { CheckCircle, Error, Close } from "@material-ui/icons";
+import { withRouter } from "react-router";
+
+let openSnackbarFn;
+
+class SnackbarMessage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      isSuccess: false,
+      status: "",
+      message: "",
+      pathname: ""
+    };
+    this.setOpen = this.setOpen.bind(this);
+    this.setIsSuccess = this.setIsSuccess.bind(this);
+    this.setStatus = this.setStatus.bind(this);
+    this.setMessage = this.setMessage.bind(this);
+    this.setPathname = this.setPathname.bind(this);
+    this.addSnackbarProperties = this.addSnackbarProperties.bind(this);
+    this.openSnackbar = this.openSnackbar.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  componentDidMount() {
+    openSnackbarFn = this.openSnackbar;
+  }
+
+  handleClose(event, reason) {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setOpen(false);
```

Comment:
Simplify with
```
if (reason !== "clickaway") {
  this.setOpen(false);
}
```

Reply:
I opted not to do this because it is convention to write the handleClose() function in this way:
![image](https://user-images.githubusercontent.com/33813045/113444989-f6e3e100-93c2-11eb-83ee-312c41dedd0e.png)

Lines 24-34 from this material-ui example code: https://github.com/mui-org/material-ui/blob/master/docs/src/pages/components/snackbars/CustomizedSnackbars.js

Reply:
I feel like that's less a convention and more of the way that guy wrote it. I would still be in favor of changing it to my suggestion provided it still works.